pytorch_lightning/plugins/precision/apex_amp.py

```diff
@@ -157,7 +157,7 @@ def pre_optimizer_step(
         # apex amp does not support closures.
         lambda_closure()
 
-        if not pl_module.automatic_optimization:
+        if pl_module.automatic_optimization:
```

Comment:
```suggestion
    def pre_optimizer_step(
        self, pl_module: LightningModule, optimizer: Optimizer, optimizer_idx: int, lambda_closure: Callable, **kwargs
    ) -> bool:
        """
        always called before the optimizer step.
        """
        # apex amp does not support closures.
        lambda_closure()

        if not pl_module.automatic_optimization:
            pl_module.trainer.call_hook("on_after_backward")
        
        optimizer.step()

        return False
```

Reply:
The pl_module.trainer.call_hook("on_after_backward") hook is already being called elsewhere in automatic_optimization.